pkg/csi_driver/controller.go

```diff
@@ -868,9 +868,11 @@ func (s *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateSn
 	}
 
 	// Check for existing snapshot
-	backupUri, _, err := file.CreateBackupURI(filer, req.Name, util.GetBackupLocation(req.GetParameters()))
+	backupLocation := util.GetBackupLocation(req.GetParameters())
+	backupUri, _, err := file.CreateBackupURI(filer, req.Name, backupLocation)
 	if err != nil {
-		return nil, err
+		klog.Errorf("Failed to create backup URI from given name %s and location %s, error: %v", req.Name, backupLocation, err.Error())
+		return nil, status.Error(codes.InvalidArgument, err.Error())
```

Comment:
My question is that: are ALL errors produced by `CreateBackupURI` invalidArgument?

Reply:
Given that BackupURI takes the name, and the params, its fair to mark it as invalidArg (unless we have a bug in parsing logic itself).